DEVELOPING.md

```diff
@@ -0,0 +1,138 @@
+# Builder Services Development Environment
+
+## Overview
+
+This document captures the steps to start and run a Builder environment for development. The builder environment includes the builder services, as well as the depot web site.
+
+There are several ways of creating a Builder dev environment - but supporting all operating systems and environments has proven to be untenable. This document includes one officially supported way of creating a Builder dev environment, and links to unsupported ways of creating the dev environment that you may use at your own risk.
+
+## Pre-Requisites
+
+### Ports required
+1. 9636 - Intra-supervisor communication
+1. 80 - Web app
+1. 9631 - supervisor api port
+1. 5432 - posgres port
+
+### Checkout
+* If you are developing on Linux
+* * Ensure you have curl
+* * `git clone https://github.com/habitat-sh/builder.git /src/builder`
+* If you are developing on a Mac
+* * `git clone https://github.com/habitat-sh/builder.git`
+
+If you are using Linux environment you can run `/src/support/linux/provision.sh` to configure your host
+If you are on a Mac, you will need brew, direnv, habitat, and Docker for Mac
+
+### GitHub OAuth Application
+`APP_HOSTNAME` mentioned below, will typically be `localhost`
+
+1. [Setup a GitHub application](https://github.com/settings/apps/new) for your GitHub organization
+1. Set the value of `Homepage URL` to `http://${APP_HOSTNAME}`
+1. Set the value of `User authorization callback URL` to `http://${APP_HOSTNAME}/` (The trailing `/` is *important*)
+1. Set the value of `Webhook URL` to `http://${APP_HOSTNAME}/`
+1. Set Repository metadata, Repository administration, Repository content and Organization members to read only (this is only used for your org so it's safe)
+1. Save and download the private key. It will result in a file like `app-name.date.private-key.pem`
+1. Copy the private key to `${HABITAT_SRC_ROOT}/.secrets/builder-github-app.pem`
+1. Record the the client-id, client-secret, app_id and public page link (in the left sidebar). These will be used for the `GITHUB_CLIENT_ID`, `GITHUB_CLIENT_SECRET`, `GITHUP_APP_ID` and `GITHUB_PUBLIC_LINK` build variables (seen below).
+
+### Create app env file
+
+1. Create `${HABITAT_SRC_ROOT}/.secrets/habitat-env`
+1. Add
+```
+  APP_HOSTNAME=localhost
+  GITHUB_API_URL=https://api.github.com
+  GITHUB_WEB_URL=https://github.com
+  GITHUB_CLIENT_ID=${GITHUB_CLIENT_ID}
+  GITHUB_CLIENT_SECRET=${GITHUB_CLIENT_SECRET}
+  GITHUB_APP_ID=${GITHUP_APP_ID}
+  GITHUB_APP_NAME="${GITHUB_PUBLIC_LINK}"
+```
+1. Save and close
+
+### Studio requirements
+
+1. Ensure you have run `hab setup` in your environment that will be executing the studio
```

Comment:
I wonder if it would make sense to move this command to `provision.sh`. It has to run before `.envrc` gets sourced or else `HAB_AUTH_TOKEN` won't be set

Reply:
`hab setup` is interactive so I'd say probably not. Also, `hab setup` only needs to be run once. Adding it to the provision script would get pretty annoying for users (like myself) that provision several times a day.

Reply:
Done. See `.envrc`